lib/fft.dx

```diff
@@ -80,25 +80,26 @@ def power_of_2_fft {log2_n}
 
   case direction of
     ForwardFT -> ans
-    InverseFT -> ans / (i_to_f n)
+    InverseFT -> ans / (n_to_f n)
 
-def pad_to_power_of_2 {a n} (log2_m:Int) (pad_val:a) (xs:n=>a) :
+def pad_to_power_of_2 {a n} (log2_m:Nat) (pad_val:a) (xs:n=>a) :
     ((Fin log2_m)=>(Fin 2))=>a =
   flatsize = intpow2 log2_m
   padded_flat = pad_to (Fin flatsize) pad_val xs
   unsafe_cast_table ((Fin log2_m)=>(Fin 2)) padded_flat
-  
+
 def convolve_complex {n m} (u:n=>Complex) (v:m=>Complex) : ((n|m)=>Complex) =
   -- Convolve by pointwise multiplication in the Fourier domain.
 
   -- Pad and convert to Fourier domain.
-  min_convolve_size = (size n) + (size m) - 1
+  -- TODO: is this use of `unsafe_nat_diff` actually valid?
```

Comment:
It doesn't seem valid to me! Perhaps use the clamping sub?

Reply:
I agree that this function has a bug where it would fail if given two size-zero arrays.  It looks like `Nat` is already starting to pay for itself!